tcms/testplans/views.py

```diff
@@ -253,28 +254,29 @@ def calculate_stats_for_testplans(plans):
     return plans
 
 
-def get(request,  # pylint: disable=missing-permission-required
-        plan_id, slug=None, template_name='plan/get.html'):
-    """Display the plan details."""
+class TestPlanGetView(DetailView):  # pylint: disable=missing-permission-required
 
-    try:
-        test_plan = TestPlan.objects.select_related().get(plan_id=plan_id)
-    except ObjectDoesNotExist:
-        raise Http404
+    template_name = 'plan/get.html'
+    http_method_names = ['get']
+    model = TestPlan
 
-    if slug is None:
-        return HttpResponsePermanentRedirect(reverse('test_plan_url',
-                                                     args=[plan_id, slugify(test_plan.name)]))
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        confirm_status_name = TestCaseStatus.get_confirmed()
+        self.object.review_case = self.object.case.exclude(case_status__name=confirm_status_name)
+        context['run_case_count'] = self.object.case.filter(case_status__name=confirm_status_name)\
+            .count
 
-    # Initial the case counter
-    confirm_status_name = 'CONFIRMED'
-    test_plan.run_case = test_plan.case.filter(case_status__name=confirm_status_name)
-    test_plan.review_case = test_plan.case.exclude(case_status__name=confirm_status_name)
+        return context
 
-    context_data = {
-        'test_plan': test_plan,
-    }
-    return render(request, template_name, context_data)
+
+class GetTestPlanRedirectView(TestPlanGetView):
```

Comment:
There is no reason why this should inherit from TestPlanGetView

Reply:
I did it because I want to use the model from the DetailView. And that way I pointed out that the two classes are related somehow.

Reply:
well the 2 views are not at all related, they just operate over the same model.